### ISSUE TITLE
Ura Quest Inside Ganon's Castle is broken

### ISSUE THREAD

**Issue by athendrix** [Admentus64/Patcher64Plus-Tool#33]
The Ura Quest version of Inside Ganon's Castle crashes on entering the first room. (The stairs)

**Comment by GhostlyDark** [comment]
Can't reproduce. Could you share a log file? Was it played on real hardware, using GLideN64 or Wii VC?

**Comment by Admentus64** [comment]
Please follow our template when writing an issue. Not enough info to go by. Make sure you used v17.1.0.

**Comment by Admentus64** [comment]
Could you also be more specific about which dungeon this is about? Inside Ganon's Castle or the stairs leading up to Ganondorf (Ganon's Tower)? In the case of the latter, it has nothing to do with Ura Quest.

**Comment by athendrix** [comment]
Sorry. Used v17.1.0
I was testing using Retroarch with the Mupen64 Plus plugin.
Inside Ganon's Castle (immediately after the rainbow bridge)
Give me a few minutes and I'll get you a log file, and everything, and I'll make sure it happens in GlideN64 too.

**Comment by athendrix** [comment]
**********************
Windows PowerShell transcript start
Start time: 20220213155013
Username: PHOENIX\Root
RunAs User: PHOENIX\Root
Configuration Name: 
Machine: PHOENIX (Microsoft Windows NT 10.0.18363.0)
Host Application: C:\windows\system32\windowspowershell\v1.0\powershell.exe -ExecutionPolicy Bypass -file C:\Users\Root\Downloads\Patcher64+ Tool\Patcher64+ Tool.ps1
Process ID: 7656
PSVersion: 5.1.18362.1474
PSEdition: Desktop
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.18362.1474
BuildVersion: 10.0.18362.1474
CLRVersion: 4.0.30319.42000
WSManStackVersion: 3.0
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
**********************
Transcript started, output file is C:\Users\Root\Downloads\Patcher64+ Tool\Files\Logs\2022-02-13-15-50-13.log
Version:       v17.1.0
Version Date:  2022-02-09
Full Path:     C:\Users\Root\Downloads\Patcher64+ Tool
System Date:   2022-02-13-15-50-12
Temp Folder:   C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Temp
Ready to patch!
ROM Path:      C:\Users\Root\Downloads\Roms\Legend of Zelda, The - Ocarina of Time (USA).n64

--- Start Patch Info ---
Game Mode:     Ocarina of Time
Console Mode:  N64
Patch Options: Ocarina of Time
Custom Header:
Patch File:
Use Options:   True
Use Redux:     False
Redux File:    Decompressed\redux.bps
Language File:  (en)
Output Name:   _patched
Command:
Downgrade:     False
Finalize:      False
Decompress:    True
ROM Hash:
No-Intro Hash: 5BD1FE107BF8106B2AB6650ABECD54D6
Wii VC Mode:   False
ROM Path:      C:\Users\Root\Downloads\Roms\Legend of Zelda, The - Ocarina of Time (USA).n64
Inject Path:
Patch Path:
Patch ROM:     True
ROM Size:      32MB
--- End Patch Info ---



--- Start Misc Settings Info ---
Ignore Input Checksum: False
Interace GUI Mode:     3
Force Show Options:    False
Use Local Temp Folder: False
--- End Misc Settings Info ---



--- Start Additional Options Info ---
 Other . MapSelect -> Translate and Enable
 Other . SkipIntro -> Skip Logo and Title Screen
 Hero Mode . MonsterHP -> 1 Monster HP
 Hero Mode . MiniBossHP -> 1 Mini-Boss HP
 Hero Mode . BossHP -> 1 Boss HP
 Dungeons . Select
 Title Screen Logo . UraQuestLogo
 Select - Dungeons . InsideTheDekuTree -> Vanilla
 Select - Dungeons . DodongosCavern -> Vanilla
 Select - Dungeons . InsideJabuJabusBelly -> Vanilla
 Select - Dungeons . ForestTemple -> Vanilla
 Select - Dungeons . FireTemple -> Vanilla
 Select - Dungeons . WaterTemple -> Vanilla
 Select - Dungeons . ShadowTemple -> Vanilla
 Select - Dungeons . SpiritTemple -> Vanilla
 Select - Dungeons . IceCavern -> Vanilla
 Select - Dungeons . BottomOfTheWell -> Vanilla
 Select - Dungeons . GerudoTrainingGround -> Vanilla
 Select - Dungeons . InsideGanonsCastle -> Ura Quest
 English Dialogue . Vanilla
--- End Additional Options Info ---

START PATCHING PROCESS OF SELECTED GAME

Converting ROM from Byteswapped to Big Endian...
Decompressing Ocarina of Time ROM...
Generated DMA Table from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Temp\converted
Decompressed ROM: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Temp\decompressed
Downgrading ROM...
ROM is already downgraded
Patching Ocarina of Time Additional Options Patches...
Patching Ocarina of Time Additional Options...
00B9FD90 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\map_select.bin
00A94994 -> Change values: 00 00 00 00 AE 08 00 14 34 84 B9 2C 8E 02 00 18 24 0B 00 00 AC 8B 00 00
00B67395 -> Change values: B9 E4 00 00 BA 11 60 80 80 09 C0 80 80 37 20 80 80 1C 14 80 80 1C 14 80 80 1C 08
00B9DAAC -> Change values: 00 00 00 00
00B17237 -> Change values: 30
00C83647 -> Multiplied values by: 00
00C83817 -> Multiplied values by: 00
00C836AB -> Multiplied values by: 00
00C5F69C -> Multiplied values by: 00
00CAAF9C -> Multiplied values by: 00
00C55A78 -> Multiplied values by: 00
00CB1BCB -> Multiplied values by: 00
00CB1903 -> Multiplied values by: 00
00CB2DD7 -> Multiplied values by: 00
00D76A07 -> Multiplied values by: 00
00C5FC3F -> Multiplied values by: 00
00C693CC -> Multiplied values by: 00
00EB797C -> Multiplied values by: 00
00C2B183 -> Multiplied values by: 00
00C2B1F7 -> Multiplied values by: 00
00CD724F -> Multiplied values by: 00
00EDC597 -> Multiplied values by: 00
00C1097C -> Multiplied values by: 00
00CD582C -> Multiplied values by: 00
00C2DEE7 -> Multiplied values by: 00
00C2DF4B -> Multiplied values by: 00
00CC6CA7 -> Multiplied values by: 00
00CC6CAB -> Multiplied values by: 00
00C96A5B -> Multiplied values by: 00
00C96B0C -> Multiplied values by: 00
00C15814 -> Multiplied values by: 00
00EEF780 -> Multiplied values by: 00
00C11177 -> Multiplied values by: 00
00C599BC -> Multiplied values by: 00
00CE60C4 -> Multiplied values by: 00
00C6471B -> Multiplied values by: 00
00C51A9F -> Multiplied values by: 00
00D74393 -> Multiplied values by: 00
00C2F97F -> Multiplied values by: 00
00C0DEF8 -> Multiplied values by: 00
00C0B804 -> Multiplied values by: 00
00CA85DC -> Multiplied values by: 00
00DADBAF -> Multiplied values by: 00
00D463BF -> Multiplied values by: 00
00BFADAB -> Multiplied values by: 00
00D09283 -> Multiplied values by: 00
00CDE1FC -> Multiplied values by: 00
00C3452F -> Multiplied values by: 00
00C3453B -> Multiplied values by: 00
00ED80EB -> Multiplied values by: 00
00EBC8B7 -> Multiplied values by: 00
00CF2667 -> Multiplied values by: 00
00DEF87F -> Multiplied values by: 00
00D49F50 -> Multiplied values by: 00
00DE9A1B -> Change values: 01
00DEB367 -> Change values: 01
00000001 -> Change values:
Value is negative!
00C44F2B -> Multiplied values by: 00
00C486CC -> Change values: 00 00 00 00
00D258BB -> Multiplied values by: 00
00D25B0B -> Multiplied values by: 00
00D64EFB -> Multiplied values by: 00
00D6223F -> Multiplied values by: 00
00C3B9FF -> Multiplied values by: 00
00CE6D2F -> Multiplied values by: 00
00D3B4A7 -> Multiplied values by: 00
00DAC824 -> Multiplied values by: 00
00D7FDA3 -> Multiplied values by: 00
00C91F8F -> Change values: 04
00CAFF33 -> Change values: 03
00E82AFB -> Change values: 04
00E87F2F -> Change values: 03
01795300 -> Patch file from: C:\Users\Root\Downloads\Patcher64+ Tool\Files\Games\Ocarina of Time\Textures\Logo\ura_logo.bin
017AE300 -> Patch file from: C:\Users\Root\Downloads\Patcher64+ Tool\Files\Games\Ocarina of Time\Textures\Logo\ura_copyright.bin
00E6E266 -> Change values: C8 96 34 21 C8
00E6E2A6 -> Change values: 64 32 35 8C 64
00E6DE2E -> Change values: 96
Patching Ocarina of Time Master Quest Dungeons...
Patching MQ Dungeon: Inside Ganon's Castle
0000CCC0 -> Change values: 03 18 E0 00 03 1A F9 20 03 18 E0 00
0318E000 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\scene.zscene
00B71544 -> Change values: 03 18 E0 00 03 1A F9 20
0000CCD0 -> Change values: 03 1A F9 20 03 1B 8B B0 03 1A F9 20
031AF920 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_0.zmap
0000CCE0 -> Change values: 03 1B 8B B0 03 1D 55 C0 03 1B 8B B0
031B8BB0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_1.zmap
0000CCF0 -> Change values: 03 1D 55 C0 03 1D CD C0 03 1D 55 C0
031D55C0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_2.zmap
0000CD00 -> Change values: 03 1D CD C0 03 1E 68 70 03 1D CD C0
031DCDC0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_3.zmap
0000CD10 -> Change values: 03 1E 68 70 03 1E E5 00 03 1E 68 70
031E6870 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_4.zmap
0000CD20 -> Change values: 03 1E E5 00 03 1F 5A 40 03 1E E5 00
031EE500 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_5.zmap
0000CD30 -> Change values: 03 1F 5A 40 03 20 57 A0 03 1F 5A 40
031F5A40 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_6.zmap
0000CD40 -> Change values: 03 20 57 A0 03 20 D4 30 03 20 57 A0
032057A0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_7.zmap
0000CD50 -> Change values: 03 20 D4 30 03 21 64 A0 03 20 D4 30
0320D430 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_8.zmap
0000CD60 -> Change values: 03 21 64 A0 03 22 1D 30 03 21 64 A0
032164A0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_9.zmap
0000CD70 -> Change values: 03 22 1D 30 03 22 89 40 03 22 1D 30
03221D30 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_10.zmap
0000CD80 -> Change values: 03 22 89 40 03 22 F2 A0 03 22 89 40
03228940 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_11.zmap
0000CD90 -> Change values: 03 22 F2 A0 03 23 A6 10 03 22 F2 A0
0322F2A0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_12.zmap
0000CDA0 -> Change values: 03 23 A6 10 03 24 23 50 03 23 A6 10
0323A610 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_13.zmap
0000CDB0 -> Change values: 03 24 23 50 03 24 C6 70 03 24 23 50
03242350 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_14.zmap
0000CDC0 -> Change values: 03 24 C6 70 03 25 43 B0 03 24 C6 70
0324C670 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_15.zmap
0000CDD0 -> Change values: 03 25 43 B0 03 25 72 30 03 25 43 B0
032543B0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_16.zmap
0000CDE0 -> Change values: 03 25 72 30 03 25 F8 D0 03 25 72 30
03257230 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_17.zmap
0000CDF0 -> Change values: 03 25 F8 D0 03 26 C1 50 03 25 F8 D0
0325F8D0 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_18.zmap
0000CE00 -> Change values: 03 26 C1 50 03 27 3E 90 03 26 C1 50
0326C150 -> Patch file from: C:\Users\Root\AppData\Roaming\Patcher64+ Tool\Ocarina of Time\Extracted\Ura Quest\Inside Ganon's Castle\room_19.zmap
Compressing Ocarina of Time ROM...
Used DMA Table: 0 1 2 3 4 5 6 7 8 9 15 16 17 18 19 20 21 22 23 24 25 26 942 944 946 948 950 952 954 956 958 960 962 964 966 968 970 972 974 976 978 980 982 984 986 988 990 992 994 996 998 1000 1002 1004 1510 1511 1512 1513 1514 1515 1516 1517 1518 1519 1520 1521 1522 1523 1524 1525
Compressed ROM: C:\Users\Root\Downloads\Roms\Legend of Zelda, The - Ocarina of Time (USA)_patched.z64
Updated CRC hash for ROM: C:\Users\Root\Downloads\Roms\Legend of Zelda, The - Ocarina of Time (USA)_patched.z64
Finished patching the Ocarina of Time ROM file.
Cleaning up files...

**Comment by athendrix** [comment]
And I can confirm that a default install of Project64 with the default settings (GLideN64) crashes on entering the first room of Inside Ganon's Castle with this error:
```
2022/02/13 17:21:54.830 13560,Error,User Interface,CNotificationImp::DisplayError,Fatal Error: Stopping emulation
```
And this happens even if you load the "Stairs to Ganon" (Which loads fine) and then go backwards to outside the inner tower, and then try and go through the door back to the exit.

**Comment by athendrix** [comment]
If you need, I can also create a patch file from the original ROM and the patched version, and post it here.
Xdelta or IPS or whatever you want.

**Comment by athendrix** [comment]
After further testing, and narrowing it down, it seems that it's exactly the combination of changing "Monster HP" in any way, and the Ura Quest version of "Inside Ganon's Castle"

**Comment by Admentus64** [comment]
I see... I go confirm this.

**Comment by Admentus64** [comment]
I can confirm there was an issue with properly setting the HP value for the Iron Knuckle, which is present in the first room of Ganon's Castle in both MQ and Ura. This issue only existed in 1 HP mode, and not default HP or HP multiply mode. This issue will be fixed with v17.1.1. It's likely this was the cause of the issue you were having, but this has to be confirmed further.

**Comment by Admentus64** [comment]
Ok. Found the cause for the issue. It's the 1 HP value for the Armos, which is also present in this first room in both MQ and Ura.

**Comment by Admentus64** [comment]
HP values have no affect on the Armors, so it will be excluded from OoT and MM with v17.1.1. That should solve this issue.

**Comment by Admentus64** [comment]
Should be fixed with v17.1.1 now
